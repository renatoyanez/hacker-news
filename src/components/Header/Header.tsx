import { ReactNode } from 'react';
import styles from './header.module.css';

interface HeaderProps {
	title: ReactNode | ReactNode[];
}

const Header = ({ title }: HeaderProps) => {
	return <div className={styles.header}>{title}</div>;
};

export default Header;
