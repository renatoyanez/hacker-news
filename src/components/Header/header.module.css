.header {
	padding: 2.75rem 0 2.625rem 9.375rem;
	box-shadow: 0 1px 4px 0 rgba(0, 21, 41, 0.12);
	background-image: linear-gradient(
		to bottom,
		#ececec -32%,
		#fff 124%
	);
	position: relative;
}
