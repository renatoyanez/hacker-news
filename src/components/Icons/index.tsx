export { default as Like } from './Like';
export { default as Clock } from './Clock';
