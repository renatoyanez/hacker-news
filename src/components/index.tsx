export { default as Dropdown } from './Dropdown';
export { default as Header } from './Header';
export { default as Layout } from './Layout';
export { default as NewsListSection } from './NewsListSection';
export { default as Tabs } from './Tabs';
export { default as Typography } from './Typography';
