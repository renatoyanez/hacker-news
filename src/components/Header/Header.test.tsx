import React from 'react';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import Header, { HeaderProps } from './Header';
import { Typography } from 'components';

const text = 'Header title';

const Title = () => <Typography>{text}</Typography>;

const props: HeaderProps = { title: <Title /> };

describe('Header', () => {
	it('renders correctly', () => {
		const tree = create(<Header {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders with props', async () => {
		const { getByText } = render(<Header {...props} />);

		const placeholder = getByText(text);
		expect(placeholder).toBeTruthy();
	});
});
