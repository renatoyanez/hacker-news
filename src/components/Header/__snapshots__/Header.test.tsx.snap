// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="header"
>
  <p
    className="textStyle"
  >
    Header title
  </p>
</div>
`;
